### PR TITLE
Remove hotplug torture script loop

### PR DESCRIPTION
**Pull request by valschneider**

### CODE REVIEW

File: lisa/target_script.py [hunk 1]
```diff
@@ -112,6 +112,7 @@ def run(self, as_root=False, background=False, timeout=None):
             if background:
                 self._bg_shell = self._target.background(self._remote_path,
                                                     as_root=self._run_as_root)
+                return self._bg_shell
```
Comment (valschneider) [inline lisa/target_script.py]: That's quite ugly, maybe we should go for separate `run()` & `run_background()` methods

**Comment by deggeman** [comment]
With these two patches in place, the hotplus tests worked  for 50 times (w/o reboot) on Juno and TC2 (TC2 needs the new TC2 topic branch w/ "arm: Use common outgoing-CPU-notification code" (2019-01-29 Paul E. McKenney).

File: lisa/target_script.py (hunk 1 above)
Comment (douglas-raillard-arm) [reply lisa/target_script.py]: I don't think that's an issue since nobody is relying on the return value currently (as it is None).

**Comment by valschneider** [comment]
I changed things around a bit and took the opportunity to cleanup `TargetScript`.

File: lisa/tests/hotplug/torture.py [hunk 2]
```diff
@@ -108,15 +109,14 @@ def cpuhp_seq(cls, nr_operations, hotpluggable_cpus, max_cpus_off, random_gen):
 
     @classmethod
     def _cpuhp_script(cls, te, res_dir, sequence, sleep_min_ms,
-                             sleep_max_ms, timeout_s, random_gen):
+                             sleep_max_ms, random_gen):
         """
         Generate a script consisting of a random sequence of hotplugs operations
 
         Two consecutive hotplugs can be separated by a random sleep in the script.
         The hotplug stress must be stopped after some time using the timeout_s
```
Comment (qperret) [inline lisa/tests/hotplug/torture.py]: That needs updating

File: lisa/target_script.py [hunk 3]
```diff
@@ -89,43 +92,73 @@ def push(self):
...
+
+        :param as_root: Kill the script as root
+        :type as_root: bool
+        """
+        if not self._proc:
+            raise RuntimeError('No background process currently executing')
+
+        cmd_pid = '$(pgrep {})'.format(self.script_name)
+        self.target.kill(cmd_pid, as_root=as_root)
+        if self._proc:
```
[the remark this answers is not in the document]
Comment (valschneider) [reply lisa/target_script.py]: Not anymore...

File: lisa/target_script.py [hunk 4]
```diff
@@ -89,43 +92,73 @@ def push(self):
...
         """
         Run the previously pushed script
+
+        :param as_root: Execute that script as root
+        :type as_root: bool
+
+        :param timeout: Timeout (in seconds) for the execution of the script
+        :type timeout: int
+        """
+        self._prerun_check()
```
[the remark this answers is not in the document]
Comment (valschneider) [reply lisa/target_script.py]: Mmm so I was about to do this but then remembering why I separated `push()` from `run()` in the first place - we want to have the script pushed first, then we can enable tracing/energy measurement and start running the script.

`lisa.wlgen.rta.RTA` pushes the rt-app file at `__init__()`, and the actual execution is done in `run()`. I was thinking we could perhaps move the target script to wlgen (it is after all a workload) but it's a peculiar one, as we have the command funnel hackery thing going on (for which I have another PR incoming so it's not a dead feature).

[the remark this answers is not in the document]
Comment (douglas-raillard-arm) [reply lisa/target_script.py]: yes, following offline discussion, let's keep `push()` as it is.

**Comment by valschneider** [comment]
@douglas-raillard-arm I let `background()` return the `Popen` handle and added a note to point out how it should be used.

**Pull request merged**